### ISSUE TITLE
lotus-miner RPC call to 'Filecoin.NetPeers':  method not supported

### ISSUE THREAD

**Issue by jennijuju** [filecoin-project/lotus#6969]
### Checklist

- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] I **am** reporting a bug w.r.t one of the [M1 tags](https://github.com/filecoin-project/lotus/discussions/6852#discussioncomment-1043951). If not, choose another issue option [here](https://github.com/filecoin-project/lotus/issues/new/choose).
- [X] I am **not** reporting a bug around deal making. If yes, create a [M1 Bug Report For Deal Making](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fbug%2CM1-release&template=m1_bug_report_deal.yml).
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.

### Lotus component

lotus JSON-RPC API

### Lotus Tag and  Version

```text
m1.3.3
```

### Describe the Bug

A miner who is on m1.3.3, has split the market process is seeing `Filecoin.NetPeers': method not supported` in Miner/Sealing/Storage lotus-miner log about every 60 seconds and the other sees it ~ every 10 seconds.

Not to sure how things are tied together rnow but wild guess is this is caused by calling `lotus-miner info all` which calls NetPeers via `GetCommonAPI` -> get repo API, which will be the full node api [here](https://github.com/filecoin-project/lotus/blob/9e707e18e11f46b609e541e471ad8f1112ea690b/node/modules/storageminer.go#L209)? And NetPeers is not one of the APIs.

This likely needs more analysis.

### Logging Information

```text
2021-07-31T14:11:01.356-0400	WARN	rpc	go-jsonrpc@v0.1.4-0.20210217175800-45ea43ac2bec/handler.go:279	error in RPC call to 'Filecoin.NetPeers': method not supported:
    github.com/filecoin-project/lotus/api.init
        /home/tbtminer/lotus/api/proxy_gen.go:42
```
```


### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by raulk** [comment]
> Not to sure how things are tied together rnow but wild guess is this is caused by calling lotus-miner info all which calls NetPeers via GetCommonAPI -> get repo API, which will be the full node api here? And NetPeers is not one of the APIs.

I think that's precisely right. Do you have a script or job that calls `lotus-miner info all` periodically, such that it justify the reocurrence?

**Comment by jennijuju** [comment]
> > Not to sure how things are tied together rnow but wild guess is this is caused by calling lotus-miner info all which calls NetPeers via GetCommonAPI -> get repo API, which will be the full node api here? And NetPeers is not one of the APIs.
> 
> I think that's precisely right. Do you have a script or job that calls `lotus-miner info all` periodically, such that it justify the reocurrence?

@meatball13 ^

**Comment by jennijuju** [comment]
> > Not to sure how things are tied together rnow but wild guess is this is caused by calling lotus-miner info all which calls NetPeers via GetCommonAPI -> get repo API, which will be the full node api here? And NetPeers is not one of the APIs.
> 
> I think that's precisely right. Do you have a script or job that calls `lotus-miner info all` periodically, such that it justify the reocurrence?

One of the two reports has confirmed, yes.
But either way: 
- I assuming NetPeers is now only callable by market process if MRA is what’s causing this? And the rationale would be you won’t need peer info for your miner process?
- But I personally think the main lotus-miner cli should not be limited by what’s subsystem is enabled . For example, I’d expect when calling “lotus-miner info all” it fetches the data from the subsystem, if any enabled and return me the result.
- @raulk we might have a bigger issue where we are breaking ecosystem tools by introducing “api breaking” changes(is this right to say?) to storageminer interface

**Comment by Meatball13** [comment]
No scripts or anything that's calls that command.  Even to the point where I shut down everything other than the miner and the Miner/Seal/Storage lotus-miner process and the error is still being called.  I could very well have a misconfiguration somewhere in my bashrc or config.toml files, but after digging through everything I can't find anything that I'm making that call every minute.

Is there anything within the code itself that makes a net-peers call on a regular basis?

**Comment by Meatball13** [comment]
One quick addition.   Running just straight 'lotus-miner net peers'  throws an "ERRROR: method not supported" unless I throw in 'lotus-miner --call-on-markets net peers', which then works successfully.  Since the error is showing up in the log of the Mining/Sealing/Storage miner, maybe there's something in my config or the code that's making that call to the regular miner and not the markets miner?

**Comment by Meatball13** [comment]
So, just want to follow up with this item as I'm still seeing the error pop up every 60 seconds.  I did want to verify that I do have a watch script running that is calling lotus-miner info, but that's set to only run once every 15 minutes.  And if I kill that script or restart the miner fresh, the error continues to pop up.  Clearly I've gotta have something messed up in one of my config.toml's or one of the other configs related to the market split, but I can't seem to pin it down.

**Comment by Meatball13** [comment]
So there's only one thing that I can possibly see that's happening every 60 seconds, and that's some processing by the Daemon related to challenges and proofs.  So, every 60 seconds I still continue to see the following error showing in my Mining/Sealing/Storage miner log:

```
2021-08-05T19:55:01.183-0400	WARN	rpc	go-jsonrpc@v0.1.4-0.20210217175800-45ea43ac2bec/handler.go:279	error in RPC call to 'Filecoin.NetPeers': method not supported:
    github.com/filecoin-project/lotus/api.init
        /home/tbtminer/lotus/api/proxy_gen.go:42 
```

I don't know if it's related, but right before that, every 60 seconds, my daemon logs show the following:

```
2021-08-05T19:55:00.329-0400	INFO	chain	chain/sync_manager.go:232	selected sync target: [bafy2bzacecfeoa6zbhexntln5k7muwb5wprpx5hzfmwr7gklgdicrjdpk5cqu]
2021-08-05T19:55:00.329-0400	INFO	chain	chain/sync_manager.go:314	worker 17174 syncing in [bafy2bzacecfeoa6zbhexntln5k7muwb5wprpx5hzfmwr7gklgdicrjdpk5cqu]
2021-08-05T19:55:00.337 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: start
2021-08-05T19:55:00.337 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:start
2021-08-05T19:55:00.337 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:finish
2021-08-05T19:55:00.337 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: finish
2021-08-05T19:55:00.337 INFO filcrypto::proofs::api > verify_winning_post: start
2021-08-05T19:55:00.337 INFO filecoin_proofs::api::winning_post > verify_winning_post:start
2021-08-05T19:55:00.337 INFO filecoin_proofs::caches > trying parameters memory cache for: WINNING_POST[68719476736]-verifying-key
2021-08-05T19:55:00.337 INFO filecoin_proofs::caches > found params in memory cache for WINNING_POST[68719476736]-verifying-key
2021-08-05T19:55:00.342 INFO filecoin_proofs::api::winning_post > verify_winning_post:finish
2021-08-05T19:55:00.342 INFO filcrypto::proofs::api > verify_winning_post: finish
2021-08-05T19:55:00.371-0400	INFO	chain	chain/sync_manager.go:232	selected sync target: [bafy2bzacea6ind2t5qmdxq4trcxm3cdebbotm2mdkykkrlowb2wrj3usmlah4]
2021-08-05T19:55:00.371-0400	INFO	chain	chain/sync_manager.go:314	worker 17175 syncing in [bafy2bzacea6ind2t5qmdxq4trcxm3cdebbotm2mdkykkrlowb2wrj3usmlah4]
2021-08-05T19:55:00.376-0400	INFO	chain	chain/sync_manager.go:232	selected sync target: [bafy2bzaceazqptdoktfc65fxjshsptzcmkdkv4537unvne4ig32zuac7sfioq]
2021-08-05T19:55:00.376-0400	INFO	chain	chain/sync_manager.go:314	worker 17176 syncing in [bafy2bzaceazqptdoktfc65fxjshsptzcmkdkv4537unvne4ig32zuac7sfioq]
2021-08-05T19:55:00.388 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: start
2021-08-05T19:55:00.388 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:start
2021-08-05T19:55:00.388 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:finish
2021-08-05T19:55:00.388 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: finish
2021-08-05T19:55:00.389 INFO filcrypto::proofs::api > verify_winning_post: start
2021-08-05T19:55:00.389 INFO filecoin_proofs::api::winning_post > verify_winning_post:start
2021-08-05T19:55:00.389 INFO filecoin_proofs::caches > trying parameters memory cache for: WINNING_POST[68719476736]-verifying-key
2021-08-05T19:55:00.389 INFO filecoin_proofs::caches > found params in memory cache for WINNING_POST[68719476736]-verifying-key
2021-08-05T19:55:00.393 INFO filecoin_proofs::api::winning_post > verify_winning_post:finish
2021-08-05T19:55:00.393 INFO filcrypto::proofs::api > verify_winning_post: finish
2021-08-05T19:55:00.407 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: start
2021-08-05T19:55:00.407 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:start
2021-08-05T19:55:00.407 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:finish
2021-08-05T19:55:00.407 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: finish
2021-08-05T19:55:00.409-0400	INFO	chain	chain/sync_manager.go:232	selected sync target: [bafy2bzacecblvvsbt776537fq7wykuti7fvukrfao75ewbwypbbeimveetsue]
2021-08-05T19:55:00.409-0400	INFO	chain	chain/sync_manager.go:314	worker 17177 syncing in [bafy2bzacecblvvsbt776537fq7wykuti7fvukrfao75ewbwypbbeimveetsue]
2021-08-05T19:55:00.410 INFO filcrypto::proofs::api > verify_winning_post: start
2021-08-05T19:55:00.410 INFO filecoin_proofs::api::winning_post > verify_winning_post:start
2021-08-05T19:55:00.410 INFO filecoin_proofs::caches > trying parameters memory cache for: WINNING_POST[34359738368]-verifying-key
2021-08-05T19:55:00.411 INFO filecoin_proofs::caches > found params in memory cache for WINNING_POST[34359738368]-verifying-key
2021-08-05T19:55:00.417 INFO filecoin_proofs::api::winning_post > verify_winning_post:finish
2021-08-05T19:55:00.417 INFO filcrypto::proofs::api > verify_winning_post: finish
2021-08-05T19:55:00.440-0400	INFO	chain	chain/sync_manager.go:232	selected sync target: [bafy2bzaceafwdwz32rboxr4obzu2petkiitynvn5szswz7urbgvyvud4fdde2]
2021-08-05T19:55:00.440-0400	INFO	chain	chain/sync_manager.go:314	worker 17178 syncing in [bafy2bzaceafwdwz32rboxr4obzu2petkiitynvn5szswz7urbgvyvud4fdde2]
2021-08-05T19:55:00.447 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: start
2021-08-05T19:55:00.447 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:start
2021-08-05T19:55:00.447 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:finish
2021-08-05T19:55:00.447 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: finish
2021-08-05T19:55:00.448 INFO filcrypto::proofs::api > verify_winning_post: start
2021-08-05T19:55:00.448 INFO filecoin_proofs::api::winning_post > verify_winning_post:start
2021-08-05T19:55:00.448 INFO filecoin_proofs::caches > trying parameters memory cache for: WINNING_POST[34359738368]-verifying-key
2021-08-05T19:55:00.448 INFO filecoin_proofs::caches > found params in memory cache for WINNING_POST[34359738368]-verifying-key
2021-08-05T19:55:00.452 INFO filecoin_proofs::api::winning_post > verify_winning_post:finish
2021-08-05T19:55:00.452 INFO filcrypto::proofs::api > verify_winning_post: finish
2021-08-05T19:55:00.462-0400	INFO	chain	chain/sync.go:700	block validation	{"took": 0.130449432, "height": "996710", "age": 0.462899623}
2021-08-05T19:55:00.465-0400	INFO	chainstore	store/store.go:646	New heaviest tipset! [bafy2bzacecfeoa6zbhexntln5k7muwb5wprpx5hzfmwr7gklgdicrjdpk5cqu] (height=996710)
2021-08-05T19:55:00.466-0400	INFO	chain	chain/sync_manager.go:322	worker 17174 done; took 136.863254ms
2021-08-05T19:55:00.506-0400	INFO	chain	chain/sync.go:700	block validation	{"took": 0.128076917, "height": "996710", "age": 0.506894942}
2021-08-05T19:55:00.508-0400	INFO	chainstore	store/store.go:646	New heaviest tipset! [bafy2bzacea6ind2t5qmdxq4trcxm3cdebbotm2mdkykkrlowb2wrj3usmlah4 bafy2bzacecfeoa6zbhexntln5k7muwb5wprpx5hzfmwr7gklgdicrjdpk5cqu] (height=996710)
2021-08-05T19:55:00.509-0400	INFO	chain	chain/sync_manager.go:322	worker 17175 done; took 138.010833ms
2021-08-05T19:55:00.515-0400	INFO	chain	chain/sync.go:700	block validation	{"took": 0.135278541, "height": "996710", "age": 0.515761178}
2021-08-05T19:55:00.518-0400	INFO	chainstore	store/store.go:646	New heaviest tipset! [bafy2bzacea6ind2t5qmdxq4trcxm3cdebbotm2mdkykkrlowb2wrj3usmlah4 bafy2bzaceazqptdoktfc65fxjshsptzcmkdkv4537unvne4ig32zuac7sfioq bafy2bzacecfeoa6zbhexntln5k7muwb5wprpx5hzfmwr7gklgdicrjdpk5cqu] (height=996710)
2021-08-05T19:55:00.519-0400	INFO	chain	chain/sync_manager.go:322	worker 17176 done; took 142.734914ms
2021-08-05T19:55:00.550-0400	INFO	chain	chain/sync.go:700	block validation	{"took": 0.136700216, "height": "996710", "age": 0.550076188}
2021-08-05T19:55:00.551-0400	INFO	chainstore	store/store.go:646	New heaviest tipset! [bafy2bzacea6ind2t5qmdxq4trcxm3cdebbotm2mdkykkrlowb2wrj3usmlah4 bafy2bzaceazqptdoktfc65fxjshsptzcmkdkv4537unvne4ig32zuac7sfioq bafy2bzacecblvvsbt776537fq7wykuti7fvukrfao75ewbwypbbeimveetsue bafy2bzacecfeoa6zbhexntln5k7muwb5wprpx5hzfmwr7gklgdicrjdpk5cqu] (height=996710)
2021-08-05T19:55:00.552-0400	INFO	chain	chain/sync_manager.go:322	worker 17177 done; took 143.802863ms
2021-08-05T19:55:00.558 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: start
2021-08-05T19:55:00.558 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:start
2021-08-05T19:55:00.558 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:finish
2021-08-05T19:55:00.558 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: finish
2021-08-05T19:55:00.559 INFO filcrypto::proofs::api > verify_winning_post: start
2021-08-05T19:55:00.559 INFO filecoin_proofs::api::winning_post > verify_winning_post:start
2021-08-05T19:55:00.559 INFO filecoin_proofs::caches > trying parameters memory cache for: WINNING_POST[34359738368]-verifying-key
2021-08-05T19:55:00.559 INFO filecoin_proofs::caches > found params in memory cache for WINNING_POST[34359738368]-verifying-key
2021-08-05T19:55:00.563 INFO filecoin_proofs::api::winning_post > verify_winning_post:finish
2021-08-05T19:55:00.563 INFO filcrypto::proofs::api > verify_winning_post: finish
2021-08-05T19:55:00.564-0400	INFO	chain	chain/sync.go:700	block validation	{"took": 0.121789726, "height": "996710", "age": 0.564501306}
2021-08-05T19:55:00.566-0400	INFO	chainstore	store/store.go:646	New heaviest tipset! [bafy2bzacea6ind2t5qmdxq4trcxm3cdebbotm2mdkykkrlowb2wrj3usmlah4 bafy2bzaceazqptdoktfc65fxjshsptzcmkdkv4537unvne4ig32zuac7sfioq bafy2bzacecblvvsbt776537fq7wykuti7fvukrfao75ewbwypbbeimveetsue bafy2bzaceafwdwz32rboxr4obzu2petkiitynvn5szswz7urbgvyvud4fdde2 bafy2bzacecfeoa6zbhexntln5k7muwb5wprpx5hzfmwr7gklgdicrjdpk5cqu] (height=996710)
2021-08-05T19:55:00.568-0400	INFO	chain	chain/sync_manager.go:322	worker 17178 done; took 127.994563ms
2021-08-05T19:55:00.758-0400	INFO	net/identify	identify/id.go:376	failed negotiate identify protocol with peer	{"peer": "12D3KooWBf5oP3GiuY34owYk1YPJTdmnqWZXTrMuhMPqATH4Bhhe", "error": "stream reset"}
2021-08-05T19:55:00.770-0400	INFO	chain	chain/sync_manager.go:232	selected sync target: [bafy2bzaceajdevkw52ddhtgveh6w22fs42cjugeimwyqfpxy2axxtomzwgp36]
2021-08-05T19:55:00.770-0400	INFO	chain	chain/sync_manager.go:314	worker 17179 syncing in [bafy2bzaceajdevkw52ddhtgveh6w22fs42cjugeimwyqfpxy2axxtomzwgp36]
2021-08-05T19:55:00.784 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: start
2021-08-05T19:55:00.784 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:start
2021-08-05T19:55:00.784 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:finish
2021-08-05T19:55:00.784 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: finish
2021-08-05T19:55:00.784 INFO filcrypto::proofs::api > verify_winning_post: start
2021-08-05T19:55:00.784 INFO filecoin_proofs::api::winning_post > verify_winning_post:start
2021-08-05T19:55:00.784 INFO filecoin_proofs::caches > trying parameters memory cache for: WINNING_POST[68719476736]-verifying-key
2021-08-05T19:55:00.784 INFO filecoin_proofs::caches > found params in memory cache for WINNING_POST[68719476736]-verifying-key
2021-08-05T19:55:00.789 INFO filecoin_proofs::api::winning_post > verify_winning_post:finish
2021-08-05T19:55:00.789 INFO filcrypto::proofs::api > verify_winning_post: finish
2021-08-05T19:55:00.889-0400	INFO	chain	chain/sync.go:700	block validation	{"took": 0.116190378, "height": "996710", "age": 0.88965102}
2021-08-05T19:55:00.892-0400	INFO	chainstore	store/store.go:646	New heaviest tipset! [bafy2bzacea6ind2t5qmdxq4trcxm3cdebbotm2mdkykkrlowb2wrj3usmlah4 bafy2bzaceazqptdoktfc65fxjshsptzcmkdkv4537unvne4ig32zuac7sfioq bafy2bzacecblvvsbt776537fq7wykuti7fvukrfao75ewbwypbbeimveetsue bafy2bzaceafwdwz32rboxr4obzu2petkiitynvn5szswz7urbgvyvud4fdde2 bafy2bzacecfeoa6zbhexntln5k7muwb5wprpx5hzfmwr7gklgdicrjdpk5cqu bafy2bzaceajdevkw52ddhtgveh6w22fs42cjugeimwyqfpxy2axxtomzwgp36] (height=996710)
2021-08-05T19:55:00.893-0400	INFO	chain	chain/sync_manager.go:322	worker 17179 done; took 122.880028ms
2021-08-05T19:55:01.118 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: start
2021-08-05T19:55:01.118 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:start
2021-08-05T19:55:01.118 INFO filecoin_proofs::api::winning_post > generate_winning_post_sector_challenge:finish
2021-08-05T19:55:01.118 INFO filcrypto::proofs::api > generate_winning_post_sector_challenge: finish
```

Could there possibly be some sort of call that related to new sectors and/or chain synchs that are calling net peers against the Miner/Storage/Sealing lotus-miner process vs. the market.  It would make sense since if you call lotus-miner net peers without --call-on-markets it does throw that method not supported error.

**Comment by Meatball13** [comment]
Alright, found a solution.  Thanks to @tippyflits and @benjaminH.  Way back when I had attempted (but never succeeded) in installing lotus-farcaster.  I went in and nuked any traces of grafana, prometheus and docker and the error has gone away.  Even though I wasn't using it, something must have stuck and been running in the background and running a poll every 60 seconds.  

So, while this fixed it for me, I assume whoever uses lotus-farcaster will need to figure out how update the code to use the --call-on-markets flag for net peers or anything else that it might be using.
